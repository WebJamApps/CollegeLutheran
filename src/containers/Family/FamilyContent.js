import React from 'react';
import PropTypes from 'prop-types';
import ReactHtmlParser from 'react-html-parser';
import PicSlider from '../../components/pic-slider';

const FamilyContent = ({ familyPics }) => (
  <div className="page-content">
    <div id="familySlideshowWide" style={{ display: 'flex', flexDirection: 'column' }}>
      {familyPics.length > 0 ? (<PicSlider data={familyPics} />) : null}
      <p style={{
        display: 'flex', alignSelf: 'center', fontSize: '14px', textAlign: 'center',
      }}
      >
        A test to make sure data is being rendered correctly. Still needs to be passed to pic-slider to render all.
      </p>
    </div>
    <div className="container-fluid">
      <p style={{ fontSize: '4pt', margin: '0' }}>&nbsp;</p>
      <div className="material-content elevation3" style={{ maxWidth: '998px', paddingBottom: '-80px', margin: 'auto' }}>
        <h3 style={{ paddingBottom: '15px', fontWeight: '100' }}>Children and Families</h3>
        <p>
          <span style={{ color: 'rgb(0, 51, 102)' }}>
            <strong>
              <i>&quot;The members of the body that seem to be weaker are indispensable.&quot;</i>
              &nbsp;&nbsp;&nbsp;1 Corinthians 12:22
            </strong>
          </span>
        </p>
        <p>
            Children are indispensable at College Lutheran Church. The Children and Family Ministry at College Lutheran Church supports kids in
            preschool through 5th grade and their families. Age appropriate Sunday School, monthly meet-ups, and
            seasonal events such as Trunk or Treating, Christmas Programs, Easter Parties, and Vacation Bible School are a few of the ways that
            we familiarize our children with Biblical stories and the power of living in Christian community.
        </p>
        <p>
            Children are also encouraged to be full participants in worship. They may follow along in the bulletin with a parent or use the
            children’s activity bags located outside the doors at the back of the sanctuary. Both the children’s
            sermon
            and the kid’s food offering are special parts of the service designated for kid participation. Additionally, there
            is a Nursery available during worship (located on the 2nd floor), if parents prefer to have their children enjoy a safe
            play zone.
        </p>
        <p>
            We celebrate the milestones of Baptism and First Communion and offer guidance for parents during these momentous events. If you
            would like your child to be baptized or receive First communion, please contact Pastor David at
          {' '}
          <a href="mailto:pastordrebes@collegelutheran.org">pastordrebes@collegelutheran.org</a>
          <br />
          We invite you and your family to come experience a fun, embracing Christian community at any or all of our upcoming events.
        </p>
        <section style={{ textAlign: 'left', marginTop: '0' }}>
          {' '}
          {ReactHtmlParser(familyPics.comments)}
        </section>
      </div>
      <div style={{
        textAlign: 'center', margin: 'auto', paddingTop: '0', paddingBottom: '-40px',
      }}
      >
        <a href="http://www.elca.org/" target="_blank" rel="noopener noreferrer">
          <img
            id="elcaLogo"
            alt="ELCA LOGO"
            src="https://dl.dropboxusercontent.com/s/wkzubcmmm3pqst4/elca-logo.png?dl=0"
            style={{ width: '350px', paddingTop: '30px', margin: 'auto' }}
          />
        </a>
      </div>
    </div>
  </div>
);
FamilyContent.defaultProps = { familyPics: [] };
FamilyContent.propTypes = {
  familyPics: PropTypes.arrayOf(PropTypes.shape({
    comments: PropTypes.string,
  })),
};
export default FamilyContent;
