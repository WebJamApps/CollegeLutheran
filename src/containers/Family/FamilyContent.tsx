import React from 'react';
import ReactHtmlParser from 'react-html-parser';
import PicSlider from '../../components/PicSlider';
import ELCALogo from '../../components/elcaLogo';
import type { FamilyProps } from './index';

const pageName = 'family';// eslint-disable-next-line max-len
const additionalContent = '<hr><h5>Resources for Families</h5><p><strong><em>Roots and Wings</em></strong>, a monthly publication of the Virginia Synod that provides suggestions of daily family time activities:<a target="_blank" href="https://www.vasynod.org/ministries/roots-and-wings/">https://www.vasynod.org/ministries/roots-and-wings/</a></p><p><strong>Ideas to bring worship to the home:</strong>&nbsp;<a target="_blank" href="https://www.vasynod.org/wp-content/uploads/2012/11/Worship-in-the-Home.pdf">https://www.vasynod.org/wp-content/uploads/2012/11/Worship-in-the-Home.pdf</a></p><p><strong>Hand Prayer:</strong>&nbsp;<a target="_blank" href="http://www.vibrantfaithathome.org/item/hand-prayers">http://www.vibrantfaithathome.org/item/hand-prayers</a></p>';
const FamilyContent = ({ familyPics }: FamilyProps): JSX.Element => (
  <div className="page-content">
    <div id="familySlideshowWide" style={{ display: 'flex', flexDirection: 'column' }}>
      {familyPics && familyPics.length > 0 ? (<PicSlider data={familyPics} />) : null}
    </div>
    <div className="container-fluid">
      <p style={{ fontSize: '4pt', margin: '0' }}>&nbsp;</p>
      <div className="material-content elevation3" style={{ maxWidth: '998px', paddingBottom: '-80px', margin: 'auto' }}>
        <h3 style={{ paddingBottom: '15px' }}>Children and Families</h3>
        <p>
          <span style={{ color: 'rgb(0, 51, 102)' }}>
            <strong>
              <i>&quot;The members of the body that seem to be weaker are indispensable.&quot;</i>
              &nbsp;&nbsp;&nbsp;1 Corinthians 12:22
            </strong>
          </span>
        </p>
        <p>
          Children are indispensable at College Lutheran Church. The Children and Family Ministry at College Lutheran Church supports kids in
          preschool through 5th grade and their families. Age appropriate Sunday School, monthly meet-ups, and
          seasonal events such as Trunk or Treating, Christmas Programs, Easter Parties, and Vacation Bible School are a few of the ways that
          we familiarize our children with Biblical stories and the power of living in Christian community.
        </p>
        <p>
          Children are also encouraged to be full participants in worship. They may follow along in the bulletin with a parent or use the
          children’s activity bags located outside the doors at the back of the sanctuary. Both the children’s
          sermon
          and the kid’s food offering are special parts of the service designated for kid participation. Additionally, there
          is a Nursery available during worship (located on the 2nd floor), if parents prefer to have their children enjoy a safe
          play zone.
        </p>
        <p>
          We celebrate the milestones of Baptism and First Communion and offer guidance for parents during these momentous events. If you
          would like your child to be baptized or receive First communion, please contact Pastor David at
          {' '}
          <a href="mailto:pastordrebes@collegelutheran.org">pastordrebes@collegelutheran.org</a>
          <br />
          We invite you and your family to come experience a fun, embracing Christian community at any or all of our upcoming events.
        </p>
        <section style={{ textAlign: 'left', marginTop: '0' }}>
          {' '}
          {ReactHtmlParser(additionalContent)}
        </section>
      </div>
      <ELCALogo pageName={pageName} />
    </div>
  </div>
);
FamilyContent.defaultProps = { familyPics: [] };

export default FamilyContent;
