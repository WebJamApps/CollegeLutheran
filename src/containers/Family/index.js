import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import DefaultFamilyContent from './FamilyContent';
import mapStoreToProps from '../../redux/mapStoreToProps';
import commonUtils from '../../lib/commonUtils';

export const Family = ({ familyPics }) => {
  commonUtils.setTitleAndScroll('Children & Families');
  return (<DefaultFamilyContent familyPics={familyPics} />);
};

Family.defaultProps = { familyPics: [] };
Family.propTypes = {
  familyPics: PropTypes.arrayOf(PropTypes.shape({
    comments: PropTypes.string,
  })),
};

export default connect(mapStoreToProps, null)(Family);
