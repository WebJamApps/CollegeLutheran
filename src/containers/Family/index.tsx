import React from 'react';
import { connect } from 'react-redux';
import DefaultFamilyContent from './FamilyContent';
import mapStoreToProps, { Ibook } from '../../redux/mapStoreToProps';
import commonUtils from '../../lib/commonUtils';

export interface FamilyProps {
  familyPics?: Ibook[];
}

export const Family = ({ familyPics }: FamilyProps): JSX.Element => {
  commonUtils.setTitleAndScroll('Children & Families', window.screen.width);
  return (<DefaultFamilyContent familyPics={familyPics} />);
};

Family.defaultProps = { familyPics: [] };

export default connect(mapStoreToProps, null)(Family);
