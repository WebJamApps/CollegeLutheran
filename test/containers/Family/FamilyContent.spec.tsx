/* eslint-disable @typescript-eslint/no-explicit-any */
import renderer from 'react-test-renderer';
import { FamilySlideShow } from 'src/containers/Family/FamilyContent';

describe('FamilyContent', () => {
  it('FamilySlideShow renders correctly with images', () => {
    const familyPics = [{
      type: 'familyPics', _id: '123', url: 'https://test.com', title: 'test',
    }];
    const tree = renderer
      .create(<FamilySlideShow familyPics={familyPics} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
