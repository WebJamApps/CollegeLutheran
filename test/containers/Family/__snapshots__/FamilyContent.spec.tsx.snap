// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FamilyContent FamilySlideShow renders correctly with images 1`] = `
<div
  id="familySlideshowWide"
  style={
    {
      "display": "flex",
      "flexDirection": "column",
    }
  }
>
  <div
    id="pic-slider"
  >
    <div
      className="slick-slider slick-initialized"
    >
      <div
        className="slick-list"
      >
        <div
          className="slick-track"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onMouseOver={[Function]}
          style={
            {
              "opacity": 1,
            }
          }
        >
          <div
            aria-hidden={false}
            className="slick-slide slick-active slick-current"
            data-index={0}
            onClick={[Function]}
            style={
              {
                "left": -0,
                "opacity": 1,
                "outline": "none",
                "position": "relative",
                "transition": "opacity 5000ms ease, visibility 5000ms ease",
                "width": 0,
              }
            }
            tabIndex="-1"
          >
            <div>
              <div
                style={
                  {
                    "display": "inline-block",
                    "width": "100%",
                  }
                }
                tabIndex={-1}
              >
                <img
                  alt="test"
                  className="slide-show"
                  src="https://test.com"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
